python/state_space_estimation/estimation.py

```diff
@@ -127,14 +130,38 @@ def choose_states(self, n_states, alpha=0.05, tests=['score', 'constraint'], ret
                 results['valid'] = results['valid'].astype(bool)
         return results
 
-            
-    def choose_states_parallel(self, n_states, alpha=0.05, tests=['score', 'constraint'], return_tests=True, verbose=False):
-        '''
-        See self.choose_states; implements same functionality with a parallel backend.
-        '''
-        variables = self.data.columns.values[:np.int64(len(self.data.columns.values)/2)]
+    def choose_states_parallel(self, n_states,
+                               alpha=0.05,
+                               tests=['score', 'constraint'],
+                               return_tests=True,
+                               serial=False,
+                               verbose=False):
+        """
+        Args:
+            n_states:
+            serial: bool: whether to perform the iterations more slowly, not in parallel
+
+        Returns:
+             See self.choose_states; implements same functionality with a parallel backend.
+        """
+        opts = {
+            'tests': tests,
+            'alpha': alpha,
+            'return_tests': return_tests,
+            'verbose': verbose
+        }
+
         states = self.potential_states(n_states=n_states)
-        results = Parallel(n_jobs=cpu_count())(delayed(self.evaluate_states)(state, tests, alpha, verbose) 
-                                            for state in tqdm(states, 
-                                                                total=(nCr(len(variables), n_states) * (2 ** (n_states)))))
-        return pd.DataFrame(results)
\ No newline at end of file
+
+        # TODO: can the expression below be simpler?
+        len_variables = len(self.data.columns.values[:np.int64(len(self.data.columns.values)/2)])
+        tqdm_total = nCr(len_variables, n_states) * (2 ** n_states)
+
+        if serial:

```

Comment:
Slow mode for debugger to step through (presume this is needed?)

Reply:
I think It would be best to just use `self.choose_states` in that case, otherwise we might not catch issues that are related specifically to parallelisation. I think using `self.chose_states_parallel` with a `serial` option is a bit confusing.